lib/arjdbc/postgresql/connection_methods.rb

```diff
@@ -42,6 +42,10 @@ def postgresql_connection(config)
     properties['tcpKeepAlive'] ||= config[:keepalives] if config.key?(:keepalives)
     properties['kerberosServerName'] ||= config[:krbsrvname] if config[:krbsrvname]
 
+    # Need to force the prepareThreshold to 1 for AR unit tests otherwise
+    # we won't receive an error for the tests that check on the statement cache being invalidated
+    properties['prepareThreshold'] = 1 if database.start_with?('activerecord_unittest')
```

Comment:
let's not have code like these around, please :wink:

Reply:
@kares This property needs to be set somewhere for running AR tests otherwise there is at least one failure.  Can you think of a good way of including this as a hook only when running AR tests?

Reply:
This change fixes 2 AR tests and unfortunately breaks another, but I think that is going to be one of the tests we just have to accept as broken at this point. I'd rather put it somewhere else but I'm not familiar enough with the test setup to know where else it can be inserted.

Reply:
are we seriously talking about having `database.start_with?` check out for `adapter: postgresql` :)
you probably have some context I am missing but let's not surprise users with hack such as those.

if its much needed to running Rails tests why not put it into the test adapter/database [configuration](https://github.com/jruby/activerecord-jdbc-adapter/blob/master/test/db/postgres_config.rb) ?

Reply:
@kares this is really only meant for a setup ONLY for running AR tests.  I think it was put in here initially due to lack of better location.